subsys/net/lib/lwm2m/lwm2m_message_handling.c

```diff
@@ -2290,6 +2293,12 @@ static int handle_request(struct coap_packet *request, struct lwm2m_message *msg
 		goto error;
 	}
 
+	if (IS_ENABLED(CONFIG_LWM2M_GATEWAY_OBJ_SUPPORT)) {
+		if (lwm2m_gw_handle_req(msg) == 0) {
+				return 0;
+			}
+	}
```

Comment:
```suggestion
	if (IS_ENABLED(CONFIG_LWM2M_GATEWAY_OBJ_SUPPORT)) {
		if (lwm2m_gw_handle_req(msg) == 0) {
				return 0;
		}
	}
```

Whitespace error.

Reply:
Another thing: If you think that callbacks should not automatically respond with messages, then I think we should call here `lwm2m_reset_message(msg, true);` and then return something negative so that the `lwm2m_udp_receive()` does not assume the `msg` is now a response.

For example, if we release message, we could call `return -ENOMSG` here.